### PR TITLE
Optimize and fix events expose

### PR DESCRIPTION
**Pull request by samplise**
### What changes were proposed in this pull request?

1. Fix previous events and expose bugs. For example, not relaunch event should only exposed when there is an error and EDL does not relaunch a new worker.
2. Add detailed information in events.
3. Optimize events action, message etc.

### Why are the changes needed?

Make the events include correct and detailed information.

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Unit test.

### CODE REVIEW

File: dlrover/python/master/node/training_node.py [hunk 1]
```diff
@@ -184,6 +184,26 @@ def __init__(
         self._lock = threading.Lock()
         self._node_id_iter = itertools.count(len(self._nodes))
         self._node_rank_iter = itertools.count(len(self._nodes))
+        self._pending_nodes: List[Node] = []
+
+    @property
+    def pending_nodes(self):
+        pending_nodes = [node.name for node in self._pending_nodes]
```
Comment (BalaBalaYi) [inline dlrover/python/master/node/training_node.py:191]: ```return [node.name for node in self._pending_nodes]```

File: dlrover/python/master/node/training_node.py [hunk 2]
```diff
@@ -184,6 +184,26 @@ def __init__(
...
+        return pending_nodes
+
+    @property
+    def first_pending_node(self):
+        if len(self._pending_nodes) == 0:
+            return ""
```
Comment (BalaBalaYi) [inline dlrover/python/master/node/training_node.py:197]: ```return None```?

File: dlrover/python/master/node/training_node.py [hunk 3]
```diff
@@ -184,6 +184,26 @@ def __init__(
...
+        first_pending_node = min(
+            self._pending_nodes, key=lambda x: x.create_time  # type: ignore
+        )
+        return first_pending_node.name
+
+    @property
+    def cur_nodes(self):
+        cur_nodes = [node.name for node in self._nodes.values()]
```
Comment (BalaBalaYi) [inline dlrover/python/master/node/training_node.py:205]: ```return [node.name for node in self._nodes.values()]```

File: dlrover/python/master/node/dist_job_manager.py [hunk 4]
```diff
@@ -244,7 +244,7 @@ def should_early_stop(self):
                 ErrorMonitorConstants.TYPE_INFO,
                 "job",
                 ErrorMonitorConstants.ACTION_EARLY_STOP,
-                msg,
+                "PS OOM",
```
Comment (BalaBalaYi) [inline dlrover/python/master/node/dist_job_manager.py:247]: Use constant.

File: dlrover/python/master/elastic_training/rdzv_manager.py [hunk 5]
```diff
@@ -584,6 +670,17 @@ def report_network_check_result(
                 f"Round {self._rdzv_round}: The node elapsed time "
                 f"are {node_check_times}"
             )
+            if self._error_monitor:
+                self._error_monitor.report_event(
+                    event_type=ErrorMonitorConstants.TYPE_INFO,
+                    instance="network-check",
+                    action=ErrorMonitorConstants.ACTION_STOP,
+                    msg="",
```
Comment (BalaBalaYi) [inline dlrover/python/master/elastic_training/rdzv_manager.py:678]: give some simple msg?

File: dlrover/python/master/elastic_training/rdzv_manager.py [hunk 6]
```diff
@@ -431,6 +438,46 @@ def get_comm_world(
                     logger.info(
                         f"Node ids are {node_ids}.\n Node IPs are {node_ips}"
                     )
+                    node_elapsed_time = time.time() - self._lastcall_time
+                    if self._error_monitor:
+                        self._error_monitor.report_event(
+                            ErrorMonitorConstants.TYPE_INFO,
+                            "job",
```
[the remark this answers is not in the document]
Comment (samplise) [reply dlrover/python/master/elastic_training/rdzv_manager.py:448]: We do not have the job name here. Moreover, it is easy to obtain it in manual or automatic analysis.

File: dlrover/python/master/node/training_node.py (hunk 3 above)
Comment (samplise) [reply dlrover/python/master/node/training_node.py:205]: We could print the return value directly in this way.

File: dlrover/python/master/node/training_node.py (hunk 2 above)
Comment (samplise) [reply dlrover/python/master/node/training_node.py:196]: We could print the return value directly in this way.

**Review by BalaBalaYi — APPROVED** [review]

**Pull request merged**